contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -491,22 +495,28 @@ abstract contract PoSValidatorManager is
         if (delegator.status != DelegatorStatus.Active) {
             revert InvalidDelegatorStatus(delegator.status);
         }
+
         // Only the delegation owner or parent validator can end the delegation.
         if (delegator.owner != _msgSender()) {
             // Validators can only remove delegations after the minimum stake duration has passed.
-            if ($._posValidatorInfo[validationID].owner == _msgSender()) {
-                if (
-                    block.timestamp
-                        < validator.startedAt + $._posValidatorInfo[validationID].minStakeDuration
-                ) {
-                    revert MinStakeDurationNotPassed(uint64(block.timestamp));
-                }
-            } else {
+            if ($._posValidatorInfo[validationID].owner != _msgSender()) {
                 revert UnauthorizedOwner(_msgSender());
             }
+
+            if (
+                block.timestamp
+                    < validator.startedAt + $._posValidatorInfo[validationID].minStakeDuration
+            ) {
+                revert MinStakeDurationNotPassed(uint64(block.timestamp));
+            }
         }
 
         if (validator.status == ValidatorStatus.Active) {
+            // Check that minimum stake duration has passed.
+            if (block.timestamp < delegator.startedAt + $._minimumStakeDuration) {
```

Comment:
For my own understanding, we only check that the minimuStakeDuration has passed when ending a delegation if the validator is still active. We don't perform this check if it is not since we still want the delegator to receive their reward in that case. I agree that this is the right semantic, but couldn't an attacker still stall the validator manager if they only delegate (potentially a large amount) to validators with staking periods that are about to end?

Reply:
Staking periods won't have defined ends, but they could end any time after their minimum stake duration has elapsed. A delegator could potentially collude with a validator to get them to end immediately, but for any reasonable `_minimumStakeDuration`,  this won't be repeatable. Especially since delegators are capped at 4-5x the validator's stake.